### PR TITLE
Fix broken mochitests

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

When we removed the integration-test.js file from MC @juliandescottes found some [discrepencies][d]

[d]: https://bugzilla.mozilla.org/show_bug.cgi?id=1401659

* asm was fixed prior
*  breaking.js looks fine

### CODE REVIEW

**Comment by codehag** [comment]
We have a lot of failures -- looks like something strange is happening on the CI

File: src/test/mochitest/head.js [hunk 1]
```diff
@@ -617,10 +617,17 @@ function togglePauseOnExceptions(
  * @return {Promise}
  * @static
  */
-function invokeInTab(fnc) {
+function invokeInTab(fnc, args) {
   info(`Invoking function ${fnc} in tab`);
-  return ContentTask.spawn(gBrowser.selectedBrowser, fnc, function*(fnc) {
-    content.wrappedJSObject[fnc](); // eslint-disable-line mozilla/no-cpows-in-tests, max-len
+  return ContentTask.spawn(gBrowser.selectedBrowser, fnc, function*(fnc, args) {
+    content.wrappedJSObject.apply(func, args); // eslint-disable-line mozilla/no-cpows-in-tests, max-len
```
Comment (darkwing) [inline src/test/mochitest/head.js]: Should this be `fnc`, not `func`?

Comment (darkwing) [reply src/test/mochitest/head.js]: OK, so there are two problems here:

1.  It should be `fnc`, not `func`
2.  `TypeError: content.wrappedJSObject.apply is not a function` bricks any test that uses `invokeInTab`.  So, for example, if I dumb it way down with the following, all tests run:

```js
function invokeInTab(fnc, args) {
  info(`Invoking function ${fnc} in tab`);
  return ContentTask.spawn(gBrowser.selectedBrowser, fnc, function*(fnc, args) {
    // content.wrappedJSObject.apply(fnc, args); // eslint-disable-line mozilla/no-cpows-in-tests, max-len
    console.log('invoking in tab: ', fnc, args);

    // FIX HERE
    if(args) {
      content.wrappedJSObject[fnc](...args);
    }
    else {
      content.wrappedJSObject[fnc]();
    }
  });
}
```

File: src/test/mochitest/browser_dbg-returnvalues.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+function getValue(dbg, index) {
+  return findElement(dbg, "scopeValue", index).innerText;
+}
+
+function toggleScopes(dbg) {
+  return findElement(dbg, "scopesHeader").click();
+}
+
+async function testReturnValue(dbg, val) {
+  invokeInTab(dbg, "return_something",val);
```
Comment (darkwing) [inline src/test/mochitest/browser_dbg-returnvalues.js]: We're sending 3 args here, while `invokeInTab` only accepts two; don't think we need `dbg`.

Comment (darkwing) [reply src/test/mochitest/browser_dbg-returnvalues.js]: ...actually, simply removing that doesn't fix test failures.  Still looking into it.

File: src/test/mochitest/head.js [hunk 3]
```diff
@@ -617,10 +617,17 @@ function togglePauseOnExceptions(
...
+  });
+}
+
+function evalInTab(script) {
```
Comment (darkwing) [inline src/test/mochitest/head.js]: This function doesn't appear to be used -- can we get rid of this?

Comment (jasonLaster) [reply src/test/mochitest/head.js]: yeah, lets kill it

**Comment by jasonLaster** [comment]
@darkwing can you push some of your changes...

**Comment by jasonLaster** [comment]
Closing and opening issue #4308

File: src/test/mochitest/head.js (hunk 1 above)
Comment (jimblandy) [reply src/test/mochitest/head.js]: A call like `ContentTask.spawn(browser, arg, task)` only passes `task` a single argument, which is a structured clone of `arg`. So I think we could do something like this:
```
function invokeInTab(fnc, ...args) {
  info(`Invoking in tab: ${fnc}(${args.map(uneval).join(',')})`);
  return ContentTask.spawn(gBrowser.selectedBrowser, { fnc, args }, function*({ fnc, args }) {
    content.wrappedJSObject[fnc](...args); // eslint-disable-line mozilla/no-cpows-in-tests, max-len
  });
}

```

**Pull request closed without merging**